src/load_fund_data.py

```diff
@@ -0,0 +1,233 @@
+        date_raw (str): The raw date string.
+
+    Returns:
+        str: The cleaned date string in "YYYY-MM-DD" format.
+    """
+    date_patterns = [
+        (r"\b\d{2}-\d{2}-\d{4}\b", "%d-%m-%Y"),
+        (r"\b\d{2}-\d{2}-\d{4}\b", "%m-%d-%Y"),
+        (r"\b\d{2}_\d{2}_\d{4}\b", "%d_%m_%Y"),
+        (r"\b\d{2}_\d{2}_\d{4}\b", "%m_%d_%Y"),
+        (r"\b\d{4}-\d{2}-\d{2}\b", "%Y-%m-%d"),
+        (r"\b\d{8}\b", "%Y%m%d"),
+    ]
+    for pattern, date_format in date_patterns:
+        date_match = re.search(pattern, date_raw)
+        if date_match:
+            raw_date = date_match.group()
+            try:
+                parsed_date = datetime.strptime(raw_date, date_format)
+                normalized_date = parsed_date.strftime("%Y-%m-%d")
+                return normalized_date
+            except ValueError:
+                continue
+    return None
+
+
+def parse_filename(filename: str) -> tuple[str, str]:
+    """
+    Parses the filename to extract the fund name and date.
+    This assumes the filename is in the format "fund_name_part.date_part".
+
+    Args:
+        filename (str): The filename to parse.
+
+    Returns:
+        tuple[str, str]: A tuple containing the fund name and date.
+
+    Raises:
+        ValueError: If the filename does not contain a valid fund name or date.
+    """
+    base_file_name = os.path.splitext(filename)[0]
+    assert len(base_file_name.split(".")) == 2, f"Invalid filename format: {filename}"
+    fund_name_raw = base_file_name.split(".")[0]
+    date_raw = base_file_name.split(".")[1]
+    fund_name = clean_fund_name(fund_name_raw)
+    date = clean_date(date_raw)
+    if not fund_name or not date:
+        raise ValueError(
+            f"Invalid filename format: {filename}. Parsed fund name: {fund_name}, Parsed date: {date}"
+        )
+    return fund_name, date
+
+
+def pandera_to_polars_dtype(pa_dtype: pa.dtypes.DataType) -> pl.datatypes.DataType:
+    """
+    Map Pandera data types to Polars data types.
+    Falls back to pl.Object if mapping is unknown.
+    """
+    mapping = {
+        String: pl.String,
+        Float: pl.Float64,
+        Int: pl.Int64,
+        Bool: pl.Boolean,
+        DateTime: pl.Datetime,
+    }
+    return mapping.get(pa_dtype.__class__, pl.String)
+
+
+def append_metadata(file_path: str, fund_name: str, date: str) -> pl.DataFrame:
+    """
+    Reads the csv and appends fund_name and date as metadata.
+    """
+    df = pl.read_csv(file_path)
+    df = df.with_columns([pl.lit(fund_name).alias("FUND"), pl.lit(date).alias("DATE")])
+    df = df.rename(
+        {col: Utils.convert_to_screaming_snake_case(col) for col in df.columns}
+    )
+    expected_columns = ExternalFund.to_schema().columns.items()
+    for column_name, column_schema in expected_columns:
+        if column_name not in df.columns:
+            pl_dtype = pandera_to_polars_dtype(column_schema.dtype)
+            df = df.with_columns(pl.lit(None, dtype=pl_dtype).alias(column_name))
```

Comment:
I'm not sure what is happening here, so you should at least comment. It smells unnecessarily complicated.